test/jit/test_backends.py

```diff
@@ -167,6 +168,45 @@ def test_save_load(self):
         # Loaded module should produce the same outputs.
         self.test_execution()
 
+class BasicModuleUnavailableTest(JitBackendTestCase):
+    """
+    Tests for BasicModule with a backend that is not available.
+    Fundamentally:
+      * _jit_to_backend is successful.
+      * Execution fails with an exception.
+      * Saving is successful.
+      * Loading fails with an exception.
+    """
+
+    def setUp(self):
+        super().setUp()
+        # Create Python, JIT and backend versions of BasicModule.
+        self.module = BasicModule()
+        self.scripted_module = torch.jit.script(BasicModule())
+        self.lowered_module = torch._C._jit_to_backend(
+            "test_backend_unavailable",
+            self.scripted_module,
+            {"forward": {"": ""}},
+        )
+
+    def test_execution(self):
+        # Test execution with backend fails because the backend that is not available.
+        input = torch.randn(5)
+
+        # Test exception is thrown.
+        with self.assertRaisesRegex(Exception, r"Backend is not available."):
```

Comment:
I tried but couldn't raise RuntimeError.

E.g. if I change all the references of Exception for RuntimeError, including the resolver, I get the following errors:

ERROR: test_execution (jit.test_backends.TestBackends)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/razy/Source/mypytorch/pytorch/torch/testing/_internal/common_utils.py", line 426, in wrapper
    fn(*args, **kwargs)
  File "/Users/razy/Source/mypytorch/pytorch/test/jit/test_backends.py", line 444, in test_execution
    self.basic_module_unavailable_test.test_execution()
  File "/Users/razy/Source/mypytorch/pytorch/test/jit/test_backends.py", line 200, in test_execution
    backend_output = backend_method(*(input, input))
  File "/Users/razy/Source/mypytorch/pytorch/torch/testing/_internal/common_utils.py", line 132, in prof_meth_call
    return prof_callable(meth_call, *args, **kwargs)
  File "/Users/razy/Source/mypytorch/pytorch/torch/testing/_internal/common_utils.py", line 126, in prof_callable
    return callable(*args, **kwargs)
torch.jit.Error: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript (most recent call last):
  File "<string>", line 10, in forward
                  return _0, _1
                else:
                  raise RuntimeError("Backend is not available.")
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE

RuntimeError: RuntimeError: Backend is not available.


======================================================================
ERROR: test_save_load (jit.test_backends.TestBackends)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/razy/Source/mypytorch/pytorch/torch/testing/_internal/common_utils.py", line 426, in wrapper
    fn(*args, **kwargs)
  File "/Users/razy/Source/mypytorch/pytorch/test/jit/test_backends.py", line 451, in test_save_load
    self.basic_module_unavailable_test.test_save_load()
  File "/Users/razy/Source/mypytorch/pytorch/torch/testing/_internal/common_utils.py", line 426, in wrapper
    fn(*args, **kwargs)
  File "/Users/razy/Source/mypytorch/pytorch/test/jit/test_backends.py", line 209, in test_save_load
    imported = torch.jit.load(buffer)
  File "/Users/razy/Source/mypytorch/pytorch/torch/jit/_serialization.py", line 163, in load
    cpp_module = torch._C.import_ir_module_from_buffer(
torch.jit.Error: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript, serialized code (most recent call last):
  File "code/torch/jit.py", line 29, in __setstate__
      self.__handles = _6
    else:
      ops.prim.RaiseException(_4)
      ~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    return None
  def forward(self: torch.jit.test_backend_unavailableLoweredModule,

Traceback of TorchScript, original code (most recent call last):
  File "<string>", line 9, in __setstate__
                  self.__handles = self.__backend.compile(self.__processed_module, self.__method_compile_spec)
                else:
                  raise RuntimeError("Backend is not available.")
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE

RuntimeError: RuntimeError: Backend is not available.

Reply:
It seems like it is raised, but the class name is being included in the message?

When you use `raise Exception(...)`, what does the error message say?

Reply:
It's the same message: raise RuntimeError("Backend is not available.")

The way I changed the resolver was to still use Exception (I couldn't see a sugared value specific to RuntimeError):

    else if (name == "RuntimeError") {
      return std::make_shared<ExceptionValue>(name);
    }

The assertion just changed from Exception to RuntimeError, like this:

        with self.assertRaisesRegex(RuntimeError, r"Backend is not available."):

Do you see an issue? My guess was the resolver, but I don't know how to return a sugared value for RuntimeError.

Reply:
Is the IR different between the two cases? Based on your earlier message it seems that `raise RuntimeError(...)` is being compiled properly.

Reply:
I don't see why the IRs would be different, but I'm not sure how to check that.
Overall the code is the same, except for raising and asserting for RuntimeError vs Exception.

Reply:
> It's the same message: raise RuntimeError("Backend is not available.")
> 
> The way I changed the resolver was to still use Exception (I couldn't see a sugared value specific to RuntimeError):
> 
> ```
> else if (name == "RuntimeError") {
>   return std::make_shared<ExceptionValue>(name);
> }
> ```
> 
> The assertion just changed from Exception to RuntimeError, like this:
> 
> ```
>     with self.assertRaisesRegex(RuntimeError, r"Backend is not available."):
> ```
> 
> Do you see an issue? My guess was the resolver, but I don't know how to return a sugared value for RuntimeError.

I would think you will have to change torchscript code to do `raise RuntimeError("Backend is not available.")` instead of `raise Exception("Backend is not available.")`?